lunabot_nav/src/lunabot_nav/dstar.py

```diff
@@ -97,6 +89,7 @@ def insert(self, node: list, key: tuple):
         self.node_queue.put((key, node))
 
     # removes a given node from the queue
+    # TODO- this probably is not efficient
```

Comment:
You could replace the priority queue with a self-balanced BST to get log n removal time. I dont see an easy python library for this, and I don't think its a big deal (Dstar instead really our bottleneck here)